### PR TITLE
Add basic "build nightly wheels" instructions

### PR DESCRIPTION
**Pull request by betatim**
This adds a basic amount of instructions for how to build and upload nightly wheels for your project.

This is a start, we probably need to flesh it out a bit, but we need some kind of starting point :D

---

I sorted the authors alphabetically by surname. If there is some other sorting mechanism you prefer let me know.

### CODE REVIEW

File: spec-0005/index.md [hunk 1]
```diff
@@ -25,6 +26,34 @@ intended use-cases, and pseudo-code illustrating its use.
...
+https://github.com/scientific-python/upload-nightly-action. To use it in your "build wheels
+workflow" add the following lines as an additional step:
+```
+- name: Upload wheel 
+  uses: scientific-python/upload-nightly-wheel
+  with: 
+    artifact-names: dist/*.whl
+    anaconda-nightly-upload-token: ${secrets.UPLOAD_TOKEN}
+```
+
```
Comment (bsipocz) [inline spec-0005/index.md:56]:
```suggestion
workflow" add the following lines as an additional step:

```
- name: Upload wheel
  uses: scientific-python/upload-nightly-wheel
  with:
    artifact-names: dist/*.whl
    anaconda-nightly-upload-token: ${secrets.UPLOAD_TOKEN}
```

```

File: spec-0005/index.md [hunk 2]
```diff
@@ -25,6 +26,34 @@ intended use-cases, and pseudo-code illustrating its use.
...
+### Build nightly wheels
+
+By building and uploading wheel for your project you make it easier for projects
+that depend on you to test against your latest changes. They can then report problems
+they find.
+
+There are a few steps to implementing this for your project:
+1. Get access to the area that nightly wheels are uploaded to
+2. Setup a CI step that builds wheels for your project
+3. Setup a CI step that uploads wheels to https://anaconda.org/scientific-python-nightly-wheels/
```
Comment (bsipocz) [inline spec-0005/index.md:38]: I would reorder these and put no2 in the first position as no1 and no3 kind of belongs together

**Review by bsipocz — APPROVED** [review]
I'll commit the linter fixes, which hopefully will trigger netlify and the rendering.

File: spec-0005/index.md [hunk 3]
```diff
@@ -25,6 +26,27 @@ intended use-cases, and pseudo-code illustrating its use.
...
 -->
 
+### Build nightly wheels
+
+By building and uploading wheel for your project you make it easier for projects
+that depend on you to test against your latest changes. They can then report problems
+they find.
+
+There are a few steps to implementing this for your project:
+1. Get access to the area that nightly wheels are uploaded to
```
Comment (pllim) [inline spec-0005/index.md:37]: This needs to be elaborated. Which area? What URL?

File: spec-0005/index.md (hunk 1 above)
Comment (bsipocz) [reply spec-0005/index.md]: Hmm, something went wrong here, I'll overwrite my commit.

File: spec-0005/index.md [hunk 4]
```diff
@@ -25,6 +26,27 @@ intended use-cases, and pseudo-code illustrating its use.
...
+2. Setup a CI step that builds wheels for your project
```
Comment (pllim) [inline spec-0005/index.md:38]: Would be nice to link to some doc that teaches people how to build wheel for universal (pure Python) and specific arch (if they have Cython, C-ext, Fortran, etc). Perhaps already some guides out there? _Eyes the cookiecutters group..._

File: spec-0005/index.md [hunk 5]
```diff
@@ -25,6 +26,35 @@ intended use-cases, and pseudo-code illustrating its use.
...
+3. Setup a CI step that uploads wheels to https://anaconda.org/scientific-python-nightly-wheels/
+
+For step (1) contact XXX. Deposit the token you get as a [secret on your repository](https://docs.github.com/en/actions/security-guides/encrypted-secrets) named `UPLOAD_TOKEN`.
```
Comment (pllim) [inline spec-0005/index.md:41]: Put a placeholder so #168 can be mentioned here when it is accepted?

File: spec-0005/index.md [hunk 6]
```diff
@@ -25,6 +26,35 @@ intended use-cases, and pseudo-code illustrating its use.
...
+
+The work for step (2) depends on your project. You are probably already doing this for your
```
Comment (pllim) [inline spec-0005/index.md:43]: Actually, I am not sure if you should make this assumption. A lot of smaller project might just push their sdist tarball and call it a release. Maybe link to some external doc related to Python releases?

File: spec-0005/index.md [hunk 7]
```diff
@@ -25,6 +26,35 @@ intended use-cases, and pseudo-code illustrating its use.
...
+releases. The new thing to add is that building wheels is run on a schedule every night or
+once a week.
+
+For step (3) there is a GitHub Action that you can use. You can find the action at
+https://github.com/scientific-python/upload-nightly-action. To use it in your "build wheels
```
Comment (pllim) [inline spec-0005/index.md:48]: This looks pretty barebones. Any interest to adopt or collaborate with https://github.com/OpenAstronomy/github-actions-workflows ?

**Review by pllim — COMMENTED** [review]
Should expectation of what is a "minimally viable" wheel look like? Example: At the very least, please publish manylinux wheel and so on?

Maybe have a section that lists your core projects with wheels so people can use them as examples? Or maybe point to a package template with that info?

File: spec-0005/index.md (hunk 5 above)
Comment (betatim) [reply spec-0005/index.md:41]: I don't know what is best here. In some way there is no need for this token to be stored in a readable way. If for some reason you need the value of the token you can have it regenerated. However, people might not realise that the GitHub Secrets are read-only?

File: spec-0005/index.md (hunk 7 above)
Comment (betatim) [reply spec-0005/index.md:48]: Something for @Carreau maybe who created the GitHub action

File: spec-0005/index.md (hunk 4 above)
Comment (betatim) [reply spec-0005/index.md:38]: Do you know a good guide? I don't unfortunately.

File: spec-0005/index.md (hunk 6 above)
Comment (betatim) [reply spec-0005/index.md:43]: I don't know what good advice to give to people about this is. Should we tell people to use https://cibuildwheel.readthedocs.io/en/stable/?

File: spec-0005/index.md (hunk 4 above)
Comment (bsipocz) [reply spec-0005/index.md:38]: I would not hold up the PR on this, after all we're really focusing on the core or big libraries here, if they have complex dependencies, they should have figured out their builds already for normal releases.

File: spec-0005/index.md (hunk 7 above)
Comment (Carreau) [reply spec-0005/index.md:48]: Sure, we can look into that as well. No particular objections. Those action seem pretty complex ?

**Pull request merged**